tests/test_pdstemplates.F90

```diff
@@ -142,158 +184,302 @@ program test_pdstemplates
     if (list(i) .ne. exp_map8(i)) stop 64
   end do
   call extpdstemplate(8, list, nummap, map)
+  if (list(22) .le. 1) then
+    list(22) = 2
+  end if
+  call extpdstemplate(8, list, nummap, map)
+  if (nummap .ne. 35) stop 104
+  do i = 1, nummap
+    if (map(i) .ne. exp_extmap8(i)) stop 105
+  end do
   ! Template number 9
   call getpdstemplate(9, nummap, list, needext, iret)
   if (iret .ne. 0 .or. nummap .ne. 36 .or. .not. needext) stop 17
   do i = 1, nummap
     if (list(i) .ne. exp_map9(i)) stop 65
   end do
   call extpdstemplate(9, list, nummap, map)
+  if (list(29) .le. 1) then
+    list(29) = 2
+  end if
+  call extpdstemplate(9, list, nummap, map)
+  if (nummap .ne. 42) stop 106
+  do i = 1, nummap
+    if (map(i) .ne. exp_extmap9(i)) stop 107
+  end do
   ! Template number 10
   call getpdstemplate(10, nummap, list, needext, iret)
   if (iret .ne. 0 .or. nummap .ne. 30 .or. .not. needext) stop 18
   do i = 1, nummap
     if (list(i) .ne. exp_map10(i)) stop 66
   end do
   call extpdstemplate(10, list, nummap, map)
+  if (list(23) .le. 1) then
+    list(23) = 2
+  end if
+  call extpdstemplate(10, list, nummap, map)
+  if (nummap .ne. 36) stop 108
+  do i = 1, nummap
+    if (map(i) .ne. exp_extmap10(i)) stop 109
+  end do
   ! Template number 11
   call getpdstemplate(11, nummap, list, needext, iret)
   if (iret .ne. 0 .or. nummap .ne. 32 .or. .not. needext) stop 19
   do i = 1, nummap
     if (list(i) .ne. exp_map11(i)) stop 67
   end do
   call extpdstemplate(11, list, nummap, map)
+  if (list(25) .le. 1) then
+    list(25) = 2
+  end if
+  call extpdstemplate(11, list, nummap, map)
+  if (nummap .ne. 38) stop 110
+  do i = 1, nummap
+    if (map(i) .ne. exp_extmap11(i)) stop 111
+  end do 
   ! Template number 12
   call getpdstemplate(12, nummap, list, needext, iret)
   if (iret .ne. 0 .or. nummap .ne. 31 .or. .not. needext) stop 20
   do i = 1, nummap
     if (list(i) .ne. exp_map12(i)) stop 68
   end do
   call extpdstemplate(12, list, nummap, map)
+  if (list(24) .le. 1) then
+    list(24) = 2
+  end if
+  call extpdstemplate(12, list, nummap, map)
+  if (nummap .ne. 37) stop 112
+  do i = 1, nummap
+    if (map(i) .ne. exp_extmap12(i)) stop 113
+  end do
   ! Template number 13
   call getpdstemplate(13, nummap, list, needext, iret)
   if (iret .ne. 0 .or. nummap .ne. 45 .or. .not. needext) stop 21
   do i = 1, nummap
     if (list(i) .ne. exp_map13(i)) stop 69
   end do
   call extpdstemplate(13, list, nummap, map)
+  if (list(38) .le. 1) then
+    list(38) = 2
+  end if
+  call extpdstemplate(13, list, nummap, map)
+  if (nummap .ne. 52) stop 114
+  do i = 1, nummap
+    if (map(i) .ne. exp_extmap13(i)) stop 115
+  end do
   ! Template number 14
   call getpdstemplate(14, nummap, list, needext, iret)
   if (iret .ne. 0 .or. nummap .ne. 44 .or. .not. needext) stop 22
   do i = 1, nummap
     if (list(i) .ne. exp_map14(i)) stop 70
   end do
   call extpdstemplate(14, list, nummap, map)
+  if (list(37) .le. 1) then
```

Comment:
Not sure why you have the if statements here. Seems like you could just set list(37) = 2, without caring what it was previously set to.

Reply:
I guess I was thinking that if the maps ever changed so that the value being checked was something like 4 I wouldn't want to change it but that's unlikely to happen I suppose. I'll remove the if statements and the extra calls to extpdstemplate.